src/flags/recursion.rs

```diff
@@ -206,38 +147,43 @@ mod test {
         assert_eq!(Some(true), Recursion::enabled_from_arg_matches(&matches));
     }
 
-    #[test]
-    fn test_enabled_from_config_none() {
-        assert_eq!(None, Recursion::enabled_from_config(&Config::with_none()));
-    }
-
     #[test]
     fn test_enabled_from_config_empty() {
```

Comment:
This test should now be renamed, to make clear it is testing the `enabled_from` function, not the `enabled_from_config` function. Alternatively, extract that bit of logic from the `enabled_from` function and put it back into the `enabled_from_config` function. The same goes for the other tests of this method and the tests for `depth_from[_config]`.

Reply:
there is not more `enabled_from_config` function, so there may not be any more confusion?

Reply:
I always had the convention for test names of `test_<name of method under test>_<test case>`. Right now this breaks that convention.

Reply:
we were testing the `Recursion::enabled_from` function

`test_` `enabled_from` `config_empty`, looks like this is following the convention?

`config` before `empty` is because it's one of the functions args

Reply:
No quite correct. This test was originally testing the `Recursion::enabled_from_config` method, in the case that the passed Yaml config is empty. Now you are testing the `Recursion::enabled_from` method, with the case that no command line args were passed and that the config is empty as well. I guess we could just leave it like that, from that perspective, but to me it would not be immediately clear what this test does, just from the name. It would have been easier if there was still the `Recursion::enabled_from_config` method, to be able to test more granular.